### PR TITLE
Optimize repeated lookups of `JobResult` objects and related fields

### PR DESCRIPTION
**Pull request by jathanism**
# Closes: #1515
# What's Changed
- Added `prefetch_related` to the UI `JobResultListView.queryset` and API `JobResultViewSet.queryset` to optimize caching of related fields
- Added `get_job_content_type()` utility function to return a cached ContentType for `extras.Job` so that it can be used by callers to reduce spurious singular queries for this object

This does not yet collapse/remove the `nautobot.extras.tables.JobResultTable.related_object` column from list view, which is still resulting in extraneous `ContentType` lookups for `extras.job` for each row for some reason, but this makes a huge reduction in singular queries and greatly improves caching of `JobResult` list queries.

# TODO

- [x] Explanation of Change(s)
- [ ] Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [x] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/extras/models/jobs.py [hunk 1]
```diff
@@ -468,6 +468,9 @@ def related_name(self):
         """
         Similar to self.name, but if there's an appropriate `related_object`, use its name instead.
         """
+        if self.obj_type == get_job_content_type():
+            return self.name
+
```
Comment (jathanism) [inline nautobot/extras/models/jobs.py:473]: This is the change recommended by @u1735067

Comment (jathanism) [reply nautobot/extras/models/jobs.py:473]: Add yours to this one, thanks!

**Review by glennmatthews — APPROVED** [review]
I'm happy with the look of these changes, though I haven't done any significant performance testing of the latest diffs - do you still have that data set handy?

**Review by bryanculver — APPROVED** [review]

**Comment by jathanism** [comment]
> I'm happy with the look of these changes, though I haven't done any significant performance testing of the latest diffs - do you still have that data set handy?

As long as there are no follow-up questions: Yes.

**Pull request merged**